pytest_fixtures/core/sat_cap_factory.py

```diff
@@ -306,19 +315,31 @@ def get_deploy_args(request):
             deploy_args.update(request.param)
         else:
             deploy_args['deploy_rhel_version'] = request.param
-    return deploy_args
+    return _remove_raw_keys(deploy_args)
+
+
+def get_cap_deploy_args():
```

Comment:
And this to capsule class as a class method?

Reply:
It sounds like an interesting idea, but the Satellite/Capsule object is not initialized yet when we gather the deploy arguments. This might be worked around by using a `@classmethod` decorator. I would prefer to make this change separately after considering all the pros and cons.